app/cdap/components/GridTable/index.tsx

```diff
@@ -0,0 +1,286 @@
+  const classes = useStyles();
+
+  const [loading, setLoading] = useState(false);
+  const [headersNamesList, setHeadersNamesList] = React.useState([]);
+  const [rowsDataList, setRowsDataList] = React.useState([]);
+  const [gridData, setGridData] = useState<any>({});
+  const [missingDataList, setMissingDataList] = useState([]);
+  const [invalidCountArray, setInvalidCountArray] = useState([
+    {
+      label: 'Invalid',
+      count: '0',
+    },
+  ]);
+
+  const getWorkSpaceData = (params, workspaceId) => {
+    const gridParams = {};
+    setLoading(true);
+    DataPrepStore.dispatch({
+      type: DataPrepActions.setWorkspaceId,
+      payload: {
+        workspaceId,
+        loading: true,
+      },
+    });
+    MyDataPrepApi.getWorkspace(params)
+      .pipe(
+        flatMap((res: any) => {
+          const { dataprep } = DataPrepStore.getState();
+          console.log(res);
+          if (dataprep.workspaceId !== workspaceId) {
+            return;
+          }
+          const directives = objectQuery(res, 'directives') || [];
+          const requestBody = directiveRequestBodyCreator(directives);
+          const sampleSpec = objectQuery(res, 'sampleSpec') || {};
+          const visualization = objectQuery(res, 'insights', 'visualization') || {};
+
+          const insights = {
+            name: sampleSpec.connectionName,
+            workspaceName: res.workspaceName,
+            path: sampleSpec.path,
+            visualization,
+          };
+          requestBody.insights = insights;
+
+          const workspaceUri = objectQuery(res, 'sampleSpec', 'path');
+          const workspaceInfo = {
+            properties: insights,
+          };
+          const gridParams = {
+            directives,
+            workspaceId,
+            workspaceUri,
+            workspaceInfo,
+            insights,
+          };
+          return MyDataPrepApi.execute(params, requestBody);
+        })
+      )
+      .subscribe((response) => {
+        DataPrepStore.dispatch({
+          type: DataPrepActions.setWorkspace,
+          payload: {
+            data: response.values,
+            headers: response.headers,
+            types: response.types,
+            ...gridParams,
+          },
+        });
+        setLoading(false);
+        setGridData(response);
+      });
+  };
+
+  useEffect(() => {
+    // Get DATA from URL paramteres to get data of workspace
+    const payload = {
+      context: params.namespace,
+      workspaceId: params.wid,
+    };
+    getWorkSpaceData(payload, wid);
+  }, [wid]);
+
+  const createHeadersData = (columnNamesList: any, columnLabelsList, columnTypesList) => {
+    if (Array.isArray(columnNamesList)) {
+      return columnNamesList.map((eachColumnName) => {
+        return {
+          name: eachColumnName,
+          label: eachColumnName,
+          type: [columnTypesList[eachColumnName]],
+        };
+      });
+    }
+  };
+
+  const convertNonNullPercent = (key, nonNullValue) => {
```

Comment:
please comment this code i don't know whats going on

Reply:
Done